test/parallel/test-http2-compat-Http2ServerRequest.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const h2 = require('http2');
+
+// Http2ServerRequest should expose convenience properties
+
+const server = h2.createServer();
+server.listen(0, common.mustCall(function() {
+  server.once('request', common.mustCall(function(request, response) {
+    const expected = {
+      statusCode: null,
+      version: '2.0',
+      httpVersionMajor: 2,
+      httpVersionMinor: 0
+    };
+
+    assert.strictEqual(request.statusCode, expected.statusCode);
+    assert.strictEqual(request.httpVersion, expected.version);
+    assert.strictEqual(request.httpVersionMajor, expected.httpVersionMajor);
+    assert.strictEqual(request.httpVersionMinor, expected.httpVersionMinor);
+
+    response.stream.on('finish', common.mustCall(function() {
+      server.close();
+    }));
+    response.end(' ');
+  }));
+
+  const url = `http://localhost:${server.address().port}`;
+  const client = h2.connect(url, common.mustCall(function() {
+    const headers = {
+      ':path': '/foobar',
+      ':method': 'GET',
+      ':scheme': 'http',
+      ':authority': 'localhost'
+    };
+    const request = client.request(headers);
+    // request.on('response', common.mustCall(function(headers) {
```

Comment:
Why are these commented out? If they are todo for later, please include a `TODO(sebdeckers):` comment with a quick explanation